integration/src/main/java/ai/djl/integration/tests/nn/BlockFactoryTest.java

```diff
@@ -0,0 +1,114 @@
+import ai.djl.ndarray.NDManager;
+import ai.djl.ndarray.types.DataType;
+import ai.djl.ndarray.types.Shape;
+import ai.djl.nn.Block;
+import ai.djl.nn.BlockFactory;
+import ai.djl.nn.Blocks;
+import ai.djl.nn.SequentialBlock;
+import ai.djl.nn.SymbolBlock;
+import ai.djl.nn.core.Linear;
+import ai.djl.repository.zoo.Criteria;
+import ai.djl.repository.zoo.ModelNotFoundException;
+import ai.djl.repository.zoo.ModelZoo;
+import ai.djl.testing.Assertions;
+import ai.djl.training.initializer.Initializer;
+import ai.djl.training.util.ProgressBar;
+import ai.djl.translate.NoopTranslator;
+import ai.djl.translate.TranslateException;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import org.testng.annotations.Test;
+
+public class BlockFactoryTest {
+
+    @Test
+    public void testBlockLoadingSaving()
+            throws IOException, ModelNotFoundException, MalformedModelException,
+                    TranslateException {
+        TestBlockFactory factory = new TestBlockFactory();
+        Model model = factory.constructBlock();
+        try (NDManager manager = NDManager.newBaseManager()) {
+            Shape inputShape = new Shape(1, 3, 32, 32);
+            Block block = model.getBlock();
+            block.setInitializer(Initializer.ONES);
+            block.initialize(manager, DataType.FLOAT32, inputShape);
+            ByteArrayOutputStream os = new ByteArrayOutputStream();
+            block.saveParameters(new DataOutputStream(os));
+            ByteArrayInputStream bis = new ByteArrayInputStream(os.toByteArray());
+            Block newBlock = factory.getBlock(manager);
+            newBlock.loadParameters(manager, new DataInputStream(bis));
+            try (Model test = Model.newInstance("test")) {
+                test.setBlock(newBlock);
+                try (Predictor<NDList, NDList> predOrigin =
+                                model.newPredictor(new NoopTranslator());
+                        Predictor<NDList, NDList> predDest =
+                                test.newPredictor(new NoopTranslator())) {
+                    NDList input = new NDList(manager.ones(new Shape(1, 3, 32, 32)));
+                    NDList originOut = predOrigin.predict(input);
+                    NDList destOut = predDest.predict(input);
+                    Assertions.assertAlmostEquals(originOut, destOut);
+                }
+            }
+        }
+    }
+
+    static class TestBlockFactory implements BlockFactory {
+
+        @Override
+        public Block getBlock(NDManager manager) {
+            SequentialBlock newBlock = new SequentialBlock();
+            newBlock.add(SymbolBlock.newInstance(manager));
+            newBlock.add(Blocks.batchFlattenBlock());
+            newBlock.add(Linear.builder().setUnits(10).build());
+            return newBlock;
+        }
+
+        public Model constructBlock()
```

Comment:
I felt function name is a bit misnomer since it loads the model and modifies the last layer. 
Something like "ChangeLastBlockInModel" ?